FreeRTOS/FreeRTOS-Plus-TCP
# IPv6/multi: Reduced complexity FreeRTOS_DNS.c

Description
-----------
In FreeRTOS_DNS.c, I found 3 large and complex functions. Again, I isolated some code in static helper functions:

prvPrepareLookup()  score 57
- prvPrepare_ReadIPAddress()
- prvPrepare_CacheLookup()

prvGetHostByName()  score 71
- prvGetDNSAddress()

prvParseDNSReply()  score **507 (!)**
- prvParseDNS_ReadQuestions()
- prvParseDNS_ReadAnswers()
    - prvParseDNS_StoreAnswer()
    - prvParseDNS_StoreToCache()
- prvParseDNS_HandleLLMNRRequest()

The last function `prvParseDNSReply()` received 5 helper functions.
The difficulty with DNS is that it contains 3 name service protocols, and also it has a call-back system, that allows for asynchronous look-ups.

Test Steps
-----------
I tested the changes by doing many look-ups, both synchronous, and asynchronous. Also I tested the LLMNR and NBNS protocols.


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
